crates/ruff_linter/src/checkers/ast/analyze/expression.rs

```diff
@@ -1380,6 +1380,9 @@ pub(crate) fn expression(expr: &Expr, checker: &mut Checker) {
             if checker.enabled(Rule::DoubleNegation) {
                 flake8_simplify::rules::double_negation(checker, expr, *op, operand);
             }
+            if checker.enabled(Rule::FalsyDictGetFallback) && *op == ruff_python_ast::UnaryOp::Not {
+                ruff::rules::falsy_dict_get_fallback(checker, operand);
```

Comment:
I think we should hook up the rule in the `CallExpr` visiter only and instead use `checker.semantic().in_boolean_test()` to determine if the expression is in a boolean test position. That also ensures that it just works for `while`, match arms, assert, etc. See `len_test` for a similar rule https://github.com/astral-sh/ruff/blob/56ae73a925a6545a714c303e7aaeed4ea5b385bc/crates/ruff_linter/src/rules/pylint/rules/len_test.rs#L72